subprojects/dependency-management/src/main/java/org/gradle/internal/component/local/model/RootLocalComponentMetadata.java

```diff
@@ -67,19 +69,27 @@ public BuildableLocalConfigurationMetadata addConfiguration(String name, String
         }
 
         public void enableLocking() {
-            this.canBeLocked = true;
+            this.locked = true;
         }
 
         @Override
         void maybeAddGeneratedDependencies(ImmutableList.Builder<LocalOriginDependencyMetadata> result) {
-            if (canBeLocked) {
-                for (DependencyConstraint dependencyConstraint : dependencyLockingProvider.findLockedDependencies(getName())) {
-                    ModuleComponentSelector selector = DefaultModuleComponentSelector.newSelector(
-                        dependencyConstraint.getGroup(), dependencyConstraint.getName(), dependencyConstraint.getVersionConstraint());
-                    result.add(new LocalComponentDependencyMetadata(getComponentId(), selector, getName(), getAttributes(), null,
-                        Collections.<IvyArtifactName>emptyList(), Collections.<ExcludeMetadata>emptyList(), false, false, true, true, dependencyConstraint.getReason()));
+            if (locked) {
```

Comment:
:o: I am confused as to why we need both `locked` and `isLockDefined`.

Reply:
It is really about the different state of the system:
* Locking enabled - yes / no
* Lock file exists - yes / no
* Lock file contains modules - yes / no

And I need all these differences at different places.
But it is true that here, I could be more lax as I only care about applying the constraints iff locking is enabled. I do not care much about the exist vs. empty state.

Reply:
I see, I would rename the variables/methods for clarity, then.